### DISCUSSION TITLE
Cors behavour?

### DISCUSSION THREAD

**Discussion by ChillFish8** [poem-web/poem#151] [Q&A] [answered]
I've been playing around accessing an API created with poem which has to go through CORS.

The issue I've ran into is with the following setup of cors:

```rust
            Cors::new()
                .allow_origins(["http://127.0.0.1:3000", "http://localhost:3000"])
                .allow_header("*")
                .allow_methods([Method::GET, Method::POST, Method::DELETE, Method::OPTIONS])
                .allow_credentials(true)
```

But when trying to access an endpoint with the required authorization. The preflight check gets rejected with `401 Unauthorized` which of course causes a CORS failure on the browser.

Is this intended? Im not sure if I'm missing some setting on the cors middleware or something with CORS but this feels slightly un-natural.

**Comment by sunli829** [comment, marked as answer]
remove this line:

`.allow_header("*")`

**Reply by ChillFish8** [reply]
😓 Yup that would do it. 🙏 Thank you
